lib/Doctrine/ORM/EntityManager.php

```diff
@@ -652,9 +659,13 @@ public function refresh($entity)
      * @return void
      *
      * @throws ORMInvalidArgumentException
+     *
+     * @deprecated This method is deprecated and will be removed in Doctrine 3.0.
      */
     public function detach($entity)
     {
+        @trigger_error('Method ' . __METHOD__ . '() is deprecated and will be removed in Doctrine 3.0, please migrate to ' . self::class . '::clear().', E_USER_DEPRECATED);
```

Comment:
Linebreak here

Reply:
Also, `EntityManager#clear()` has a different behaviour... and I'm not even sure if we should point people to do partial cleaning of the UoW, it might have the very same issues we have with `EntityManager#detach()`.